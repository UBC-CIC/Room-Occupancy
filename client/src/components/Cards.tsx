import * as React from "react";
import { Card, Icon, Image, SemanticICONS } from "semantic-ui-react";
import "../App.css";

export interface ICardsProps {
  imageURL: string;
  header: string;
  subheader: string;
  description: string;
  subDescription: string | null;
  icon: SemanticICONS | undefined;
  url: string;
}

export function Cards(props: ICardsProps) {
  const {
    header,
    imageURL,
    subheader,
    description,
    subDescription,
    icon,
    url,
  } = props;
  return (
    <Card>
      <Image src={imageURL} wrapped ui={false} />
      <Card.Content>
        <Card.Header>{header}</Card.Header>
        <Card.Meta>
          <span className="date">{subheader}</span>
        </Card.Meta>
        <Card.Description>{description}</Card.Description>
      </Card.Content>
      <Card.Content extra>
        <a href={url} target="_blank">
          <Icon name={icon} />
          {subDescription}
        </a>
      </Card.Content>
    </Card>
  );
}
