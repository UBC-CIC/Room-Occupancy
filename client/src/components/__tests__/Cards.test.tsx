import React from "react";
import { render, screen } from "@testing-library/react";
import { Cards } from "../Cards";
import "@testing-library/jest-dom";
import { ICardsProps } from "../Cards";

const mockCardProps: ICardsProps = {
  imageURL: "example.jpg",
  header: "Card Header",
  subheader: "Subheader",
  description: "Description",
  subDescription: "Sub Description",
  icon: undefined,
  url: "https://example.com",
};

test("matches snapshot", () => {
  const { asFragment } = render(<Cards {...mockCardProps} />);

  expect(asFragment()).toMatchSnapshot();
});

test("renders Cards component", () => {
  render(<Cards {...mockCardProps} />);
  const cardElement = screen.getByText(mockCardProps.header);
  expect(cardElement).toBeInTheDocument();
});

test("displays image", () => {
  render(<Cards {...mockCardProps} />);
  const imageElement = screen.getByText("Card Header");
  expect(imageElement).toBeInTheDocument();
});

test("displays card header", () => {
  render(<Cards {...mockCardProps} />);
  const headerElement = screen.getByText(mockCardProps.header);
  expect(headerElement).toBeInTheDocument();
});

test("displays subheader", () => {
  render(<Cards {...mockCardProps} />);
  const subheaderElement = screen.getByText(mockCardProps.subheader);
  expect(subheaderElement).toBeInTheDocument();
});

test("displays description", () => {
  render(<Cards {...mockCardProps} />);
  const descriptionElement = screen.getByText(mockCardProps.description);
  expect(descriptionElement).toBeInTheDocument();
});
