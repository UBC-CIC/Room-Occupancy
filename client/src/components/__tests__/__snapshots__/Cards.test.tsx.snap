// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ui card"
  >
    <div
      class="image"
    >
      <img
        src="example.jpg"
      />
    </div>
    <div
      class="content"
    >
      <div
        class="header"
      >
        Card Header
      </div>
      <div
        class="meta"
      >
        <span
          class="date"
        >
          Subheader
        </span>
      </div>
      <div
        class="description"
      >
        Description
      </div>
    </div>
    <div
      class="extra content"
    >
      <a
        href="https://example.com"
        rel="noopener"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="icon"
        />
        Sub Description
      </a>
    </div>
  </div>
</DocumentFragment>
`;
